reana_client/utils.py

```diff
@@ -134,6 +136,9 @@ def _prepare_kwargs(reana_yaml):
             kwargs["parameters"] = reana_yaml.get("inputs", {}).get("parameters", {})
             kwargs["original"] = True
 
+        if workflow_type == "yadage":
+            kwargs["return_workflow_spec"] = True
+
```

Comment:
`load_reana_spec` is called upon:

- [Workflow creation](https://github.com/reanahub/reana-client/blob/732664851a6a504896e53f6346eaad8074e60c83/reana_client/cli/workflow.py#L293)
- [Workflow restart](https://github.com/reanahub/reana-client/blob/732664851a6a504896e53f6346eaad8074e60c83/reana_client/cli/workflow.py#L513) (when `--file` is passed)

Apart from validation.

So we would still get the Yadage spec returned in those cases and [sent to the server](https://github.com/reanahub/reana-client/blob/732664851a6a504896e53f6346eaad8074e60c83/reana_client/api/client.py#L133).

Reply:
@mvidalgarcia  What about this then?

`if not skip_validate_environments and workflow_type == "yadage"`?

Reply:
@audrium is working on Yadage parameters validation (no `skip_validate_env` involved), and he needs the spec loaded, so I guess it wouldn't work for his case. I see two options, both ugly, unfortunately..

- `return_yadage_spec` flag in `load_reana_spec`, `False` by default.
- What @audrium mentioned [here](https://github.com/reanahub/reana-client/pull/462#discussion_r586296218)
- Any other idea?

Reply:
Another option which is ugly as well could be that since we have created a separate `validation.py`, we can duplicate the code there and use that function to load the yadage spec just for validation.

I can't think of anything else which is not ugly.

We have to choose the least worst option.

Reply:
Just to add an example, [this](https://github.com/reanahub/reana-client/pull/470/files#diff-4f20d18bf32c181d09b016a6a363171402d113bb01347de15da4768afcbe6998R181) is what I had in mind. Still ugly solution, but nothing better comes into my mind for now

Reply:
> Just to add an example, [this](https://github.com/reanahub/reana-client/pull/470/files#diff-4f20d18bf32c181d09b016a6a363171402d113bb01347de15da4768afcbe6998R181) is what I had in mind. Still ugly solution, but nothing better comes into my mind for now

IMHO, this plus an explanatory comment would be the cleanest solution.